Homework10/Homework10/Yogurt.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Homework10
+{
+    public class Yogurt : Warehouse
+    {
```

Comment:
Your hierarchy isn't right
Warehouse can't be as a basic class for concrete product
The role of warehouse is to store of products list

I strongly recommend to edit current decision.